src/sdbus/dbus_proxy_async_interface_base.py

```diff
@@ -469,13 +422,23 @@ def new_proxy(
         return new_object
 
 
+class Closeable(Protocol):
+    def close(self) -> None:
+        ...
+
+
 class DbusExportHandle:
-    def __init__(self, local_meta: DbusLocalObjectMeta):
-        self._dbus_slots: List[SdBusSlot] = [
-            i.slot
-            for i in local_meta.activated_interfaces
-            if i.slot is not None
-        ]
+    def __init__(self, *items: Closeable) -> None:
+        self._items = list(items)
+
+    def append(self, item: Closeable) -> None:
+        self._items.append(item)
+
+    def close(self) -> None:
```

Comment:
Not sure why the `close` method is needed. The context manager is already implemented for DbusExportHandle.

https://docs.python.org/3/library/contextlib.html#contextlib.closing

> Most types managing resources support the context manager protocol, which closes thing on leaving the with statement. As such, closing() is most useful for third party types that don’t support context managers

Also `close` might be misleading because the bus objects use it to close the connections to D-Bus but here it does not actually close the D-Bus.

Reply:
- I made the DbusExportHandle a bit more generic; you can insert any type into it, which has the standard Python's close() method. Which fits nicely with the SdBus slot, which also implements this.
- Makes it more Pythonic in my opinion, as most Python types implement both the context manager protocol *and* the close() method (so it can be easily cleaned-up manually by calling the "close" method, instead of calling `__(a)exit__()` if not bound to a context)
- Not sure how close() might mislead user to thinking it closes the connection, when the method is being on an "export handle".

If you don't like it, or think the change is not worth it, I am ok with removing it. I can live without it, but it still seems to me as an improvement to the code.